### ISSUE TITLE
MercurialEclipse-2.0.0.201208040235 could not play a part in team synchronizing's perspective of eclipse 4.2 and hg version is 2.1.1

### ISSUE THREAD

**Issue by GoogleCodeExporter** [anterzhang/mercurialeclipse#49]
```
as pic


```

Original issue reported on code.google.com by `lw...@wumii.com` on 13 Aug 2012 at 12:31

Attachments:
- [1.jpg](https://storage.googleapis.com/google-code-attachments/mercurialeclipse/issue-49/comment-0/1.jpg)

**Comment by GoogleCodeExporter** [comment]
```
Were any errors shown? Can you check the error log for relevant errors or 
warnings. Window->Show View-> Other... -> Error log. Look for entries added 
when you invoke synchronize. 

Also please give more detail about what you did and what you expect to happen
```

Original comment by `johnpeb@gmail.com` on 13 Aug 2012 at 2:16

**Comment by GoogleCodeExporter** [comment]
```
there is a problem in newest 2.0.0.201208160020 version,

in the error.log:
java.lang.NullPointerException
    at java.util.Hashtable.put(Unknown Source)
    at com.vectrace.MercurialEclipse.wizards.MercurialParticipantSynchronizeWizard.createParticipant(MercurialParticipantSynchronizeWizard.java:349)
    at org.eclipse.team.ui.synchronize.ParticipantSynchronizeWizard.performFinish(ParticipantSynchronizeWizard.java:76)
    at com.vectrace.MercurialEclipse.wizards.MercurialParticipantSynchronizeWizard.performFinish(MercurialParticipantSynchronizeWizard.java:209)
    at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:827)
    at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:432)
    at org.eclipse.jface.dialogs.Dialog$2.widgetSelected(Dialog.java:624)
    at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:248)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1053)
    at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4169)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3758)
    at org.eclipse.jface.window.Window.runEventLoop(Window.java:825)
    at org.eclipse.jface.window.Window.open(Window.java:801)
    at org.eclipse.team.internal.ui.synchronize.actions.SynchronizePageDropDownAction$1.run(SynchronizePageDropDownAction.java:40)
    at org.eclipse.team.internal.ui.synchronize.actions.SynchronizePageDropDownAction.run(SynchronizePageDropDownAction.java:114)
    at org.eclipse.jface.action.Action.runWithEvent(Action.java:498)
    at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:584)
    at org.eclipse.jface.action.ActionContributionItem.access$2(ActionContributionItem.java:501)
    at org.eclipse.jface.action.ActionContributionItem$6.handleEvent(ActionContributionItem.java:452)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1053)
    at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4169)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3758)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$9.run(PartRenderingEngine.java:1022)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:916)
    at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:86)
    at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:585)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:540)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
    at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:124)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:353)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:180)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1414)

and the session data is:
eclipse.buildId=I20120608-1400
java.version=1.7.0_05
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=zh_CN
Framework arguments:  -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product 
org.eclipse.epp.package.jee.product




```

Original comment by `lw...@wumii.com` on 22 Aug 2012 at 2:51

**Comment by GoogleCodeExporter** [comment]
```
Same issue here with MercurialEclipse 2.0.1.201208192352 on Eclipse Juno
```

Original comment by `andreas....@gmail.com` on 26 Aug 2012 at 11:21

**Comment by GoogleCodeExporter** [comment]
```
Will be fixed in 2.0.1
```

Original comment by `johnpeb@gmail.com` on 3 Sep 2012 at 5:23
- Changed state: **Fixed**
